src/shared/codelens/localLambdaRunner.ts

```diff
@@ -515,16 +484,27 @@ export const invokeLambdaFunction = async (params: {
         templatePath: params.samTemplatePath,
         eventPath,
         environmentVariablePath,
-        debugPort: (params.isDebug && params.debugConfig) ? params.debugConfig.port.toString() : undefined,
-        invoker: params.samTaskInvoker
+        invoker: params.samTaskInvoker,
+        debugPort: (params.isDebug && params.debugPort) ? params.debugPort.toString() : undefined,
+        debuggerPath: params.debuggerPath
     })
 
     const startInvokeTime = new Date()
     await command.execute()
 
-    if (params.isDebug && params.debugConfig) {
+    if (params.isDebug) {
+        if (!params.debugConfig) {
+            const err = new Error('Started debug run without a debugConfig')
+            params.channelLogger.logger.error(err)
+            throw err
+        }
+        if (!params.debugPort) {
```

Comment:
This validation takes place after you've already performed a `(params.isDebug && params.debugPort)` check (line 488). This check should go before that call. The call to `command.execute()` (calls `sam local invoke`) was pointless if we get here and ultimately throw.